lib/services/auth/auth-service.js

```diff
@@ -0,0 +1,135 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var jwt = require('jsonwebtoken');
+var crypto = require('crypto');
+
+module.exports = authServiceFactory;
+di.annotate(authServiceFactory, new di.Provide('Auth.Services'));
+di.annotate(authServiceFactory,
+    new di.Inject(
+        'Services.Configuration'
+    )
+);
+function authServiceFactory(
+    configuration
+) {
+    function AuthService() {
+    }
+
+    AuthService.prototype.init = function(){
+        this.algorithm = 'HS256';
+        this.secretOrKey = 'RackHDRocks!';
+        this.ignoreTokenExpiration = false;
+        this.encoder = 'base64';
+
+        this.hashConfig = {
+            // number of bytes in the Hash
+            hashBytes: 64,
+            // number of bytes in the slat
+            saltBytes: 64,
+            // number of iterations to get the final hash, longer means more secure,
+            // but also slower
+            iterations: 10000
+        };
+
+        this.loadFromConf();
+
+        this.jwtSignOptions = {
+            algorithm : this.algorithm,
+            secretOrKey : this.secretOrKey,
+            expiresIn : this.expiresIn
+        };
+
+        this.jwtVerifyOptions = {
+            algorithm : this.algorithm,
+            secretOrKey : this.secretOrKey,
+            ignoreExpiration : this.ignoreTokenExpiration
+        };
+    }
+
+    AuthService.prototype.loadFromConf = function (){
+        this.userFromConf = configuration.get('authUsername', 'admin');
+        // Read hash and salt from configure file and decode per base64
+        this.hashFromConf = new Buffer(configuration.get('authPasswordHash',
+                'KcBN9YobNV0wdux8h0fKNqi4uoKCgGl/j8c6YGlG7iA0PB3P9ojbmANGhDlcSBE0iOTIsYsGbtSsbqP4wvsVcw=='),
+            this.encoder);
+
+        this.saltFromConf = new Buffer(configuration.get('authPasswordSalt',
+                'zlxkgxjvcFwm0M8sWaGojh25qNYO8tuNWUMN4xKPH93PidwkCAvaX2JItLA3p7BSCWIzkw4GwWuezoMvKf3UXg=='),
+            this.encoder);
+
+        if (this.hashFromConf.length !== this.hashConfig.hashBytes ||
+            this.saltFromConf.length !== this.hashConfig.saltBytes){
+            throw new Error('Invalid password hash or salt length in configure file');
+        }
+
+        this.expiresIn = configuration.get('authTokenExpireIn', 86400); //86400 = 24 * 60 * 60, 24hours
+        if (typeof (this.expiresIn) !== 'number'){
```

Comment:
FYI, you can use the assert library: `assert.number(this.expiresIn, "Invalid token expire value in configure file")`